pkg/rpc/client/rpc.go

```diff
@@ -545,3 +558,30 @@ func (c *Client) ValidateAddress(address string) error {
 	}
 	return nil
 }
+
+// CalculateValidUntilBlock calculates ValidUntilBlock field for tx using cache.
+func (c *Client) CalculateValidUntilBlock() (uint32, error) {
```

Comment:
Does it have any test?

Reply:
Added test, but it's a bit more complicated than others.